scipy/linalg/tests/test_basic.py

```diff
@@ -931,35 +930,75 @@ def test_check_finite(self):
 
 class TestDet:
     def setup_method(self):
-        np.random.seed(1234)
+        self.rng = np.random.default_rng(1680305949878959)
 
     def test_simple(self):
```

Comment:
I have some thoughts about these old tests, but I don't think you should have to overhaul old tests in this PR, If you're interested, let me know.

Reply:
Sure what do you have in mind?

Reply:
Regarding the first two tests, I'd suggest combining them and adding a comment about where the reference value comes from or, better yet, calculating it on the fly as `a * d - b * c`. 

But really, unless the tolerances of these simple tests is tightened, it seems to me that comparing randomly generated cases against `np.linalg.det` is strictly a stronger test. I would just parameteriz `test_random` over several different size arrays (e.g. `(2, 2)`, `(20, 20)`, `(3, 2 20, 20)`) and over a boolean that changes whether the arrays are complex or not. This one test could replace six.

Reply:
Ah true. I was also not so happy about the numpy dependency but let me generate some ill-conditioned examples instead so we don't need to track what happens on the numpy side.

Reply:
What's wrong with the NumPy dependency, though? These aren't testing to extreme precision, so ISTM that as long as SciPy plans to keep NumPy as a dependency and as long as NumPy keeps `np.linalg.det`, we might as well use it as a reference. This is checking our wrapper, not the numerical stability of LAPACK, right?

Reply:
LAPACK stability is highly dependent on the architecture so probably that's not what we should do given our diverse platform needs. getrf is practically tested all the way to the bits everyday so probably we will hear about it immediately. What I meant is if numpy changes its behavior, for whatever reason, then we will have failing tests for no reason. 

It's not difficult to cook up proper examples where the result should be a predictable number even the entries are not that obvious. Let me put some examples then you can check if they are sound enough.

Reply:
Now we have random parameterized ones and the known value ones just in case NumPy det breaks.